content/client-lib-development-guide/features.textile

```diff
@@ -1464,6 +1486,110 @@ enum StatsIntervalGranularity:
   DAY
   MONTH
 
+class DeviceDetails:
+  + getDeviceInfo() => io DeviceDetails
+  id: String
+  platform: DeviceType
+  type: DeviceType
+  clientId: String?
+  metadata: Dict<String, String>
+  updateTokenId: String
+  resetId(String)
+  resetUpdateToken() => io
+  push: DevicePushDetails
+
+class DevicePushDetails:
+  transportType: DevicePushTransportType
+  transportId: String
+  transportToken: String?
+  state: .Active | .Failing | .Failed
+  errorReason: ErrorInfo?
+  metadata: Dict<String, String>
+
+class Push:
+  publish(clientId: String, JsonObject) => io
+  publish(deviceId: String, JsonObject) => io
+  publish(transportType: String, transportId: String, transportTokenOrKey?: String) => io
+  admin: PushAdmin
+
+  // Only on platforms that support receiving notifications:
+
+  activate(DeviceDetails) => io ErrorInfo? // RSH2c
+  activateExternally(
```

Comment:
I think you've misundestood the intention of this method.  The `activate` comes in two forms where in the first form it is responsible for get a push token from the OS, and then registering itself with Ably via the REST API directly.  The second form it does step 2, but invokes a callback provided by the dev so that they can pass the details of that registration to their server which in turn handles that registration via the REST API.

Registration of the device via the REST API is simply done using the current `PushDeviceRegistrations#save` method.